readme.txt
My name is MINH THANH HO, I am currently studying master of Information Technology at Deakin University.
This is  the project for mobile application SIT708

The app that I am going to make call "ZBrain", it is a game that design 
to help the users to exercise their brain that contributing to
study better at school, focus at work and importantly 
maintaining psychological well-being.

There will be different level of the game, the user allows to play to the next level when they are completed the certain level. It will be select the combination of two features:
    - Logic 
    - Memory match
Firstly, the logic of the game that is designed to be natural and simple way that can suitable for most people. That involving:
    - Finding the unique item in the collection of similar things
    - Selecting to ascending/descending number
    - Multiple choice question allows the users to select the correct answer base on the knowledge
Secondly, memory match is concentration game that contains matching images with different topics with well design. The board size varies depend on the difficulty per level (2x2 tiles, 3x3 tiles, 4x4 tiles, etc.). All of the cards (images) are laid face down on a surface. In order to play the game, the user need to tap to open up to two cards at any time, and when they are a match then they stay revealed. When they are not match they will face up again. Therefore, player is required to remember cards positions to try and match all cards in the fewest steps possible. There will be the time limit and the number step allow to complete the game.
When the users finish each level, the app will be rating their performances.

Feature
    - Having board range of Logic, memory match and multiple-choice question on variety of context. 
    Most of the competitors do not provide a range of context challenge. 
    Playing with ZBrain, the users will have the great experiences on different challenges and the rich context of interaction.
    - For the type of memory match, firstly, the user will have the opportunity to view the entire board before they play the game in given time that depend on the difficulty of the game. It allows the user to reduce the number of step for solving the problem as well as improve the ability to concentration and remember card pair.
    - For the logic type game, the user first gives the chance to look at the question and decide the best solution. It helps users to have a deep thinking on the problem. ZBrain also give the hint for the users when they have difficulty. The hint will be give when they select wrong answer.
    - ZBrain has the content and challenge is researched carefully and organise according to level of difficulty
    - ZBrain give advice when the user could not able to answer the questions, so users is relaxing and enjoy the app
    - The next game and challenge will be unlocking when the users pass the current level with 3-star rating. It helps users to focus on each game and it is the value of the game.
    - The game is replayed current level when user fails to pass this level without go back to the initial level. It will give the users great experiences.
    - ZBrain is designed to allow the user to go from the easy level to challenges level so that it builds the strengthen mind.
    - Celebrating users when they completed each game. It also rates the user performances depend on how fast they solve as well as the number of step (lower number of tries are better scores). It is the great way to give the award for their achievement.
    - ZBrain allows users to share the result with friends on Facebook
    - ZBrain provide the great way to navigate the entire application without any difficulty.
    - ZBrain have portrait and landscape screen. It also cross platform apps so that give the users experience on mobile and tablet devices.
    - Play without internet, it is not a problem with ZBrain

Data Structures in gameData.js
    - gameLevel:
        It is an object data type, there are different level with the description,
        level: The level of the game
        TypeGame: Math Game, Memory Game, Order Number 
        NumberOfButtons: the number of button on the screen
        Timer: the maximum number allow for each game
        typeGame: asceding order, deceding order, yes no question, multiple choice question, drag and drop

    - numberStar: Number of Star when user finish the game, currently set to 3
    - Alphabets: array type have all  Alphabets Character.
    - equationSign: mathermatical symbol that have pluss, minus, multiple, devision array
    - backgroudMusic: different backgroudMusic file that push into the array
    - congraduation: has the different text to say congraduation to user when the finish the game
    - requireToImprove: has different text say when the user pass the game
    - encourage: has different text to encourage user. It make the user more greate experiecens to the game
    - legalData that has the object type, that has the name, licenseType, Author, Soucre. 

Major functions
    ui.js
        - createLink function
            - Create the hyperlink with give the url
            - call: createLink
            - parameter: url
            - return the div achor
        - createTable function
            - Creates a Table , with the given data 
            - parameter: data for the table
            - return the table element
        - createSoundEffect function
            - Creates a soundEffect , with the given source 
            - return the sound element
        - createParagraph function
            - Creates a paragraph , with the given text 
            - parameter: any text to include on the paragraph
            - return paragraph element
        - createHeading1,2,3 function
            - Creates a paragraph , with the given text 
            - parameter: any text to include on the heading
            - return heading 1,2,3 element
        - createDiv function
            - create Div with give the html element
            - return div element
        - createButton function
            - Creat the dom button element
            - Create the text node with give parameter
            - return button
    gameStart.js
        - displayHomePage function
            - call: displayHomePage() 
            - There is no need for parameter for that parameter
            - All element is in the div wrapper
            - Display the welcome text 
            - Display the logo image
            - Display the Navigation to the About page or to the Menu
            - Add the event to play button, When it clicked, it displays the legal page
            - Add the event to play button, When it clicked, it displays the displayGameLevelPage
    legal.js
        - displayLegal function
            - call: displayLegal()
            - The function call displayLegal which the purpose to display the 
            - legal page. 
            - It has the Navigation to to Home page.
            - Create the table to display legal session
            - createTable function by passing the legalData to the UI to display the data  
    menu.js
        - displayGameLevelPage function
            - call: displayGameLevelPage()
            - This function display the Menu
            - There is no need for parameter for that parameter
            - All element is in the div wrapper
            - Display the text to allow user to select the game level
            - Display the Navigation to the Home page and Exist to close the game
            - Display the menu that has a list of button with the stars and level
            - The menu only allow to play on the unlock level. When the user pass the current level, it will allow user to increase to the next level 
            - When the user select the particular game level, it will call the playGame to detect the type of game,
            This function will call appropirate game type.
            - No return value
        - gameLevelClick function
            - call: gameLevelClick()
            - When the user select the particular game level.
            - Parameter: levelNumber, unlockButtonElement
            - It is the anonomous function that take the level number and unlock game button
        - getBackgroundMusic function
            - call: getBackgoundMusic() 
            - Randomly select the backgournd music
            - Play the music
            - Add event to detect if the music is end
    orderNumberGame
        - displayGameLevelPage function
            - call: orderNumber(gameData) 
            - This function display order Number and alphabet
            - gameData as parameter 
            - Reset variable
            - Assign the game data to the variable
            - Display the timer
            - Create the text to tell the user what is that game about
            - Randomly select the type of game (Number and alphabet)
            - Get the random value to the button 
            - Soft the array
            - Call the anonamous function regarind to the button clicked
            - No return variable
        - runTimer function
            - call: runTimer() 
            - check the widthBar is 100, it will terminate the game
            - This function will run by interval, it will update the time bar
            - Call the endGame when the timer is end
            - Clear the interval
        - checkExistValue function
            - call: checkExistValue() 
            - parameter: newNumber which the new value to add to the array
            - return true if the value exist
            - return false if the value is not exist in the origional array
        - anonymousOrder function
            - call: anonymousOrder(number, numberButton) function, it is call when the button click
            - parameter: number which is the value, numberButton is the button element
            - Check if the user click is correct order
            - When the user already clicked the button, this button is disable
            - When the user click incorrect button, show the correct button
            - No return value
    mathGame.js
        - mathGame function
            - call: mathGame(gameData) 
            - This function display math game that including yes/no question, multiple choice question, drag and drop the mathematic symbol
            - gameData as parameter 
            - Reset variable
            - Assign the game data to the variable
            - Display the timer
            - Create the text to tell the user what is that game about
            - Check type of game (yes/No, Multiple choice, Drag Math Symbol)
            - Call the the coresponding function regarding to the type of game
            - No return value
        - displayDragMathSymbol function
            - call: displayDragMathSymbol() 
            - This function display math game drag and drop the mathematic symbol
            - No parameter 
            - In crease the question
            - updat ethe text to tell what is that game about
            - Create the array for symbol and display on the screen
            - Set the dragable to true 
            - Send the data when the symbol is drag
            - Check if the equation where first value (+ - * /) might have the same value: ex 2:1 = 2 also 2*1 = 2. Get the new equation
            - Display the equation
            - Create the div to drop the the symbol
            - Check if it it correct or not
            - play the sound effect
            - No return variable
        - displayMutipleChoiceQuestions function
            - call: displayMutipleChoiceQuestions() 
            - This function display mutiple choice questions
            - No parameter 
            - In crease the question
            - update the text to tell what is that game about
            - Call the getEquationMultiplechoiceQuestion function to display the equation
            - Create the array of the closely result
            - Swap bettwen the result
            - Create the answer div
            - Display value to the div
            - call the anonymous function when the button click passing the number and the button element
            - No return variable
        - anonymousAnswer function
            - call: anonymousAnswer() 
            - This function to display when the user click on the answer
            - guessResult mean the gues value. buttonElement is  the button element
            - Check if it is correct answer or incorrect answer
            - Change the background color
            - Call the checkCompleteQuestions function to check if the function is completed
        -  getEquationMultiplechoiceQuestion function 
            - call:getEquationMultiplechoiceQuestion()
            - no parameter 
            - Get the equation value by calling the function equationValue
            - return the json file that have the equationDiv and result 
        - getEquationYesNoQuestionGame function 
            - call:getEquationYesNoQuestionGame()
            - no parameter 
            - Get the equation value by calling the function equationValue
            - Check if the answer is true or false
            - return the json file that have the equationDiv and result 
    completedGame.js
        - getCongraduationText(stars, completed)
            - This fuction to write the congration Text randomly to make the 
            game more interesting
            - call: getCongraduationText()
            - parameter: stars to get the appropriate text, completed: to show if the user completed the game, pass or not pass
            - return text
        - endGame(stars,currentLevel,completed)  function
            - call: endGame(stars,currentLevel,completed) 
            - Parameter: stars: number of stars, currentLevel: the level of current game, completed: whereather the user pass or not pass the level
            - This function display finish game
            - Loop through arrayStars to check if the user play previous game again.Update the star
            - When the current level greater than unlockGame. Then add new game data with value.
            - Check if the currentLevel is greater than or equal unlockGame level and unlockGame must also be less the gameLevel.length. The game must completed to unlock  the next game
            - Wrap the entire the element to the wrapper
            - Display the Level name
            - Display the Level name
            - Display the text how the user perform the game
            - Display the navigation bar to comback to menu, next game if the user pass the current game, play again when the user play this game again
            - No return variable
    memoryGame.js
        - memoryGame function
            - call: memoryGame(gameData) 
            - This function display order Number and alphabet
            - gameData as parameter 
            - Reset variable
            - Assign the game data to the variable
            - Display the timer
            - Create the text to tell the user what is that game about
            - Create the memoryGameDivElement to wrap the entire game
            - Duplicate cards
            - Mix the card each time the game run
            - Loop through the Duplicate cards and 
                - Create the div card element that contain face and hide
                - Add event to the card when the card clicked
                - show card when the card is click less than two, Check if two card is matching, if it is match, hide this cards. otherwise
                    return to the previous position 
            - No return value
        - runTimer function
            - call: runTimer() 
            - check the widthBar is 100, it will terminate the game
            - This function will run by interval, it will update the time bar
            - Call the endGame when the timer is end
            - Clear the interval
        - checkClicked(clickedTarget) function
            - call: checkClicked(clickedTarget)
            - parameter: clickedTarget that the current caard clicked
            - Assing the first card cliked name
            - check if numberClicked is two then  Assign second clicked
            - No return value
        - checkAllCardReveal() function
            - call: checkAllCardReveal()
            - parameter: no
            - Check if all card reveal, then completed the game
            - No return value
        - match() function
            - call: match()
            - parameter: no
            - Get all the "picked" class. Then add class call "sameCards" 
            - Add match CSS
            - No return value

changelog.txt
Week4 [25-31/MAR]
    [Thursday 28-MAR-2019] 
        - Insert the readme.txt
        - Create all the folder neccesary for the project
        - Create the index.html to run the app
    [Sunday 31-MAR-2019]
        - Watching the video "Week 4 Make a Game"
        - Create Welcome page [continues]
        - Wrap the entire page using div and set the background color
        - Create the app logo name that hold in the div
        - Create the game data for different game level
        - when the user clicks to the play button, it will nagigate to the game Level [continues]
Week5 [2-7/APR]
    [Monday 2-APR-2019]
        - Create a botton for Memory match game that show the game board with responsive format
        - get data when the button is clicked.[continues]
    [Thursday 4-APR-2019]
        - When the user click to the play button, it will navigate to the User Level Page.
        - Create the Game Level Page gameLevelPage.js [continues]
        - Create the function to display the game level
        - create the general game level data for each level.
        - Assing the level text content to the box
        - Create the draft memory game
Week6 [8 - 14/APR]
    [Saturday 14-APR-2019]
        - Update the game level, the previously does work properly.
        - Change the game level css to make it look good (Make it to the center, change the background color and color)
        - Assign the button level handler.
        - Connect the gamelevel 1 when the user click 
        - Reorganise all the file that easy to manage.[continues]
        - When the play button click, make the return button element, assign onclick
        - Make the if-else statement to assign the game level
        - Create order number game in level three
        - Assign the data to the gameData.js  [continues]
Week7 [16 - 21/APR]
    [Tuesday 16-APR-2019]
        - on the menu, create Heading logo, assign the text menu, assign to wrapper
        - Order Number Game Level 3 [continues]
            - Create order number game and make the link in the index.html
            - Create the variable numberOfButtons and maxNumber
            - Generate the random number in the range from to maxNumber
            - Create a loop that generate through the number of buttons 
            - Display all the button on the screen.
            - assign the event to the button
            - Change the background color when the user already have clicked on the button
            - Disable the button so that it will not able to click again.
            - Acceding order array 
            - Create the if-else statement to compare if the user clicked is corrected or not
            - Create the pop-up window if the user clicked the wrong button.
            - Copy the current array number to the soft array number.
            - Add animation when the button is corrected clicked
            - Make the unique (undupplicated) origional list
    Wednesday 17-APR-2019]
        - Create the finsh game page [continues]
Trimester Break [22-28/APR]
    [Monday 22-APR-2019]
        - Add the congraduation sound when the user finish the game
        - Research how to create the timer. [continues]
Week8 [29APR - 5MAY]
    [FRriday 3-5-2019]
        - Create the game level 4 namely Memory game different version
        - Update the memory match to specify the requirement
        - Create the css namely "memoryGame" [continues]
        - Clear the sceen
        - Create the wrapper to wrap the entire the game
        - Create the data for this memory game in the data file
        - Modify the order game data in the data file
        - Link the game level to each game with the data
        - Create the checkGame function to detect the type of game. Then it will pass the data
        - Passing data to the order game number
    [Saturday 4-5-2019]
        - Update the number of stars for the game base on user performance. [continues]
        - Create the link from the completed game to the home page.
        - Create unlock Game Level as the global variable.The purpose of this variable is to identify the next unlock game.
        - Increase to the next level when the user already completed the current level.
        - Add the key symbol to the menu game. It is also disable unlock game level when this game is not completed.
        - Remember the stars (performance) of the users by create the variable arrayStars. It will show in the menu.

    [Sunday 5-5-2019]
        - Create the progess bar to make the game more intersting. The timer is set for each game that depend on the difficulty of the game.
        - Update the completed game with different text depend on the user completed the game
Week9 [6May - 12MAY]
    [Monday 6/5/2019]
        - Modify the memory game to make it easy to understand and expand. [continues]
        - Assign the backgroud image to the div and change their apperance.   
        - Add more images to the data to make more game level
        - Dupplicate the data in the memory game which allow the player to match two images.
        - Dowload and make the background music for memory game
        - Play the background music
    [Tuesday 7/5/2019]
        - Randomly to display of cards
        - Add event handler to the div
        - Add the picked class to highlight the div that have already clicked
        - Add the match class to show if two div is matching
        - Create the class to show the background transparent when two images is matching
        - Check if the user complete reveal all the cards, then complete the game
        - check if the card has previously already openned. If it is open then it effect their star (It is one of the most correct way to detect user memeorys)
Week10 [13 - 17MAY]
    [Tuesday 14/5/2019]
        - Update the UI to make the dynamic User interface that dynamic update the font-size and the width and height of the button and picture
        - Update the UI according to the landscape mode

data/gameData.js
/*
GAME LEVEL
*/
window.gameLevel = window.gameLevel || [];
gameLevel.push(
    {
        level: 1,
        typeGame: "Math Game",
        type: "Drag Math Symbol",
        maxNumber: 10,
        totalQuestions: 5,
        timer: 120
    },
    {
        level: 2,
        typeGame: "Math Game",
        type: "Multiple choice",
        maxNumber: 10,
        totalQuestions: 5,
        timer: 120
    },
    {
        level: 3,
        typeGame: "Math Game",
        type: "Yes No",
        maxNumber: 10,
        totalQuestions: 5,
        timer: 120
    },
    {
        level: 4,
        typeGame: "Math Game",
        type: "knowledge",
        totalQuestions: 4,
        questionCollection: [{
            question: "What is the greatest two digit number?",
            posibleAnswer: ["10", "20", "50", "40"],
            correctAnswer: "50"
        },
        {
            question: "Complete the sequence 13, 16, ……, 22.",
            posibleAnswer: ["17", "18", "19", "20"],
            correctAnswer: "19"
        },
        {
            question: "How many digits are there in 1000?",
            posibleAnswer: ["One digit", "Two digits", "Three digits", "Four digits"],
            correctAnswer: "Four digits"
        },
        {
            question: "What is the average value of 25, 25, 47 and 3?",
            posibleAnswer: ["23", "24", "25", "26"],
            correctAnswer: "25"
        }],
        timer: 120
    },
    {
        level: 5,
        typeGame: "Order Number",
        type: "ascending order",
        numberOfButtons: 4,
        maxNumber: 12,
        timer: 10
    },
    {
        level: 6,
        typeGame: "Order Number",
        type: "decending order",
        numberOfButtons: 4,
        maxNumber: -12,
        timer: 10
    },
    {
        level: 7,
        typeGame: "Memory Game",
        cardCollections: [{
            name: 'apple',
            img: 'images/apple.jpg'
        }, {
            name: 'cut-apple',
            img: 'images/cut-apple.jpg',
        }],
        timer: 30
    },
    {
        level: 8,
        typeGame: "Memory Game",
        cardCollections: [{
            name: 'apple',
            img: 'images/apple.jpg'
        }, {
            name: 'cut-apple',
            img: 'images/cut-apple.jpg',
        }, {
            name: 'many-apple',
            img: 'images/many-apple.jpg',
        }],
        timer: 45
    },
    {
        level: 9,
        typeGame: "Memory Game",
        cardCollections: [{
            name: 'apple',
            img: 'images/apple.jpg'
        }, {
            name: 'cut-apple',
            img: 'images/cut-apple.jpg',
        }, {
            name: 'many-apple',
            img: 'images/many-apple.jpg',
        }, {
            name: 'apple-bright.jpg',
            img: 'images/apple-bright.jpg',
        }],
        timer: 60
    },
    {
        level: 10,
        typeGame: "Order Number",
        type: "ascending order",
        numberOfButtons: 6,
        maxNumber: 20,
        timer: 40
    },
    {
        level: 11,
        typeGame: "Order Number",
        type: "decending order",
        numberOfButtons: 4,
        maxNumber: -20,
        timer: 40
    },
    {
        level: 12,
        typeGame: "Order Number",
        type: "ascending order",
        numberOfButtons: 6,
        maxNumber: 20,
        timer: 20
    },
    {
        level: 13,
        typeGame: "Order Number",
        type: "decending order",
        numberOfButtons: 6,
        maxNumber: 20,
        timer: 20
    },
    {
        level: 14,
        typeGame: "Math Game",
        type: "Drag Math Symbol",
        maxNumber: 200,
        totalQuestions: 10,
        timer: 120
    },
    {
        level: 15,
        typeGame: "Math Game",
        type: "Multiple choice",
        maxNumber: 10,
        totalQuestions: 10,
        timer: 120
    },
    {
        level: 16,
        typeGame: "Math Game",
        type: "Yes No",
        maxNumber: 10,
        totalQuestions: 10,
        timer: 120
    },
    {
        level: 17,
        typeGame: "Memory Game",
        cardCollections: [{
            name: 'apple',
            img: 'images/apple.jpg'
        }, {
            name: 'cut-apple',
            img: 'images/cut-apple.jpg',
        }, {
            name: 'many-apple',
            img: 'images/many-apple.jpg',
        }, {
            name: 'apple-bright.jpg',
            img: 'images/apple-bright.jpg',
        },
        {
            name: 'apple-hand',
            img: 'images/apple-hand.jpg',
        }],
        timer: 360
    },
    {
        level: 18,
        typeGame: "Memory Game",
        cardCollections: [{
            name: 'apple',
            img: 'images/apple.jpg'
        }, {
            name: 'cut-apple',
            img: 'images/cut-apple.jpg',
        }, {
            name: 'many-apple',
            img: 'images/many-apple.jpg',
        }, {
            name: 'apple-bright.jpg',
            img: 'images/apple-bright.jpg',
        }, {
            name: 'apple-food-fresh',
            img: 'images/apple-food-fresh.jpg',
        },
        {
            name: 'apple-hand',
            img: 'images/apple-hand.jpg',
        }],
        timer: 360
    },
    {
        level: 19,
        typeGame: "Order Number",
        type: "ascending order",
        numberOfButtons: 15,
        maxNumber: 50,
        timer: 300
    },
    {
        level: 20,
        typeGame: "Order Number",
        type: "decending order",
        numberOfButtons: 20,
        maxNumber: 100,
        timer: 300
    },
    {
        level: 21,
        typeGame: "Math Game",
        type: "Drag Math Symbol",
        maxNumber: 20,
        totalQuestions: 15,
        timer: 300
    },
    {
        level: 22,
        typeGame: "Math Game",
        type: "Multiple choice",
        maxNumber: 20,
        totalQuestions: 15,
        timer: 400
    },
    {
        level: 23,
        typeGame: "Math Game",
        type: "Yes No",
        maxNumber: 30,
        totalQuestions: 20,
        timer: 400
    },
    {
        level: 24,
        typeGame: "Order Number",
        type: "ascending order",
        numberOfButtons: 20,
        maxNumber: 200,
        timer: 900
    },
    {
        level: 25,
        typeGame: "Math Game",
        type: "knowledge",
        totalQuestions: 4,
        questionCollection: [{
            question: "What is the smallest two digit number?",
            posibleAnswer: ["10", "20", "50", "40"],
            correctAnswer: "10"
        },
        {
            question: "Complete the sequence 10, 12, ……, 16.",
            posibleAnswer: ["12", "13", "14", "15"],
            correctAnswer: "14"
        },
        {
            question: "How many digits are there in 150?",
            posibleAnswer: ["One digit", "Two digits", "Three digits", "Four digits"],
            correctAnswer: "Three digits"
        },
        {
            question: "What is the average value of 5, 10, 3 and 2",
            posibleAnswer: ["2", "3", "4", "5"],
            correctAnswer: "5"
        }],
        timer: 120
    },
    {
        level: 26,
        typeGame: "Memory Game",
        cardCollections: [{
            name: 'apple',
            img: 'images/apple.jpg'
        }, {
            name: 'cut-apple',
            img: 'images/cut-apple.jpg',
        }, {
            name: 'many-apple',
            img: 'images/many-apple.jpg',
        }, {
            name: 'apple-bright.jpg',
            img: 'images/apple-bright.jpg',
        },
        {
            name: 'apple-hand',
            img: 'images/apple-hand.jpg',
        }],
        timer: 360
    },
    {
        level: 27,
        typeGame: "Memory Game",
        cardCollections: [{
            name: 'apple',
            img: 'images/apple.jpg'
        }, {
            name: 'cut-apple',
            img: 'images/cut-apple.jpg',
        }, {
            name: 'many-apple',
            img: 'images/many-apple.jpg',
        }, {
            name: 'apple-bright.jpg',
            img: 'images/apple-bright.jpg',
        }, {
            name: 'apple-food-fresh',
            img: 'images/apple-food-fresh.jpg',
        },
        {
            name: 'apple-hand',
            img: 'images/apple-hand.jpg',
        }],
        timer: 360
    }
);

/*Number of Star for the game*/
window.numberStar = 3;

/* Alphabets Character */
var alphabets = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z'];

/* Equation Sign */
var equationSign = ['+', '-', '*', '/'];

/*background music*/
window.backgroudMusic = window.backgroudMusic || [];
backgroudMusic.push("sound/backgroundMusic1.mp3", "sound/backgroundMusic2.mp3", "sound/backgroudMusic3.mp3");

/*Text to say congraduation  */
window.congraduation = window.congraduation || [];
congraduation.push("Congratulations", "Amazing", "Perfect", "Great", "Fantastic", "Awesome", "Wonderfull");

/*Text to say pass the game with required to improve  */
window.requireToImprove = window.requireToImprove || [];
requireToImprove.push("Let's try harder next time", "Not Bad", "That alright")

/*Text to encourage the user when they must play the game again */
window.encourage = window.encourage || [];
encourage.push("Deep breathing and relaxing", "That’s okay! You will get through it next time", "Get your mind off of things and let you recharge a little bit.");

/* Legal data*/
window.legalData =
    [{
        Name: "correctAnswer.mp3",
        LicenseType: "Standard License",
        Author: "Finnolia Productions Inc",
        Source: "https://www.soundeffectsplus.com/product/correct-answer-double-01/"
    },
    {
        Name: "wrongAnswer.mp3",
        LicenseType: "Standard License",
        Author: "Finnolia Productions Inc",
        Source: "https://www.soundeffectsplus.com/product/wrong-answer-buzzer-05/"
    },
    {
        Name: "congraduationSound.mp3",
        LicenseType: "Standard License",
        Author: "Finnolia Productions Inc",
        Source: "https://www.soundeffectsplus.com/product/correct-answer-bell-gliss-01/"
    },
    {
        Name: "backgroudMusic1.mp3",
        LicenseType: "CC-BY",
        Author: "snowflake",
        Source: "http://dig.ccmixter.org/files/snowflake/59564"
    },
    {
        Name: "backgroudMusic2.mp3",
        LicenseType: "CC-BY",
        Author: "snowflake",
        Source: "http://dig.ccmixter.org/files/snowflake/58681"
    },
    {
        Name: "backgorundMisic3.mp3",
        LicenseType: "CC-BY",
        Author: "snowflake",
        Source: "http://dig.ccmixter.org/files/hansatom/54914"
    },
    {
        Name: "homeLogo.jpg",
        LicenseType: "Pexels",
        Author: "Bogdan Dirică",
        Source: "https://www.pexels.com/photo/photo-of-person-holding-crystal-ball-1645668/"
    },
    {
        Name: "apple.jpg",
        LicenseType: "Pexels",
        Author: "mali maeder",
        Source: "https://www.pexels.com/photo/healthy-apple-fruits-natural-102104/"
    },
    {
        Name: "cut-apple.jpg",
        LicenseType: "Pexels",
        Author: "John Finkelstein",
        Source: "https://www.pexels.com/photo/red-apple-1630588/"
    },
    {
        Name: "many-apple.jpg",
        LicenseType: "Pexels",
        Author: "Pixabay",
        Source: "https://www.pexels.com/photo/red-apple-fruits-162806/"
    },
    {
        Name: "apple-bright.jpg",
        LicenseType: "Pexels",
        Author: "Bruno Scramgnon",
        Source: "https://www.pexels.com/photo/apple-blur-bright-close-up-588587/"
    },
    {
        Name: "apple-food-fresh.jpg",
        LicenseType: "Pexels",
        Author: "Skitterphoto",
        Source: "https://www.pexels.com/photo/wet-green-apple-38068/"
    },
    {
        Name: "apple-hand.jpg",
        LicenseType: "Pexels",
        Author: "Aleksandar Cvetanović",
        Source: "https://www.pexels.com/photo/close-up-photo-of-person-holding-red-apple-1451649/"
    }
    ];

// MEMORY GAME
window.dragData = window.dragData || [];
// now add the data for GAME LEVEL 4 (Memory Game)
dragData.push(
    {
        name: 'apple',
        img: 'images/apple.jpg',
    },
    {
        name: 'cut-apple',
        img: 'images/cut-apple.jpg',
    },
    {
        name: 'many-apple',
        img: 'images/many-apple.jpg',
    },
    {
        name: 'apple-bright.jpg',
        img: 'images/apple-bright.jpg',
    }

);
// cardCollections.push(
//     {
//         name: 'apple',
//         img: 'images/apple.jpg',
//     },
//     {
//         name: 'cut-apple',
//         img: 'images/cut-apple.jpg',
//     },
//     {
//         name: 'many-apple',
//         img: 'images/many-apple.jpg',
//     },
//     {
//         name: 'apple-bright.jpg',
//         img: 'images/apple-bright.jpg',
//     },
//     {
//         name: 'apple-food-fresh',
//         img: 'images/apple-food-fresh.jpg',
//     },
//     {
//         name: 'apple-hand',
//         img: 'images/apple-hand.jpg',
//     },
// );


